src/org/labkey/test/util/BaseSearchHelper.java

```diff
@@ -0,0 +1,305 @@
+                    String missingResultSummary = missingResults.stream().map(Locator::toString).collect(Collectors.joining("\n"));
+                    String wrongResultSummary = missingResults.stream().map(Locator::toString).collect(Collectors.joining("\n"));
+                    errorCollector.error(baseScreenshotName + ": Incorrect search results for [\"" + searchTerm + "\"]. " +
+                            (missingResults.isEmpty() ? "" : "Missing results: \n" + missingResultSummary + "\n") +
+                            (wrongResults.isEmpty() ? "" : "Unexpected results: \n" + wrongResultSummary));
+                }
+                else
+                {
+                    notFound.add(searchTerm);
+                }
+            }
+        }
+
+        if (notFound.isEmpty())
+            TestLogger.log("All items were found");
+        else
+            TestLogger.log(notFound.size() + " items were not found.");
+
+        return notFound;
+    }
+
+    protected void addExpectedContainerLink(String expectedResultsContainer, SearchResultsQueue.SearchItem item, List<Locator> expectedResults)
+    {
+        if (expectedResultsContainer != null)
+        {
+            if ( Locator.linkContainingText("@files").findOptionalElement(getDriver()).isPresent() )
+            {
+                if(expectedResultsContainer.contains("@files"))
+                {
+                    expectedResults.add(Locator.linkWithText(expectedResultsContainer));
+                }
+                else
+                {
+                    expectedResults.add(Locator.linkWithText(expectedResultsContainer + (item.expectFileInResults() ? "/@files" : "")));
+                }
+            }
+            else
+            {
+                expectedResults.add(Locator.linkWithText(expectedResultsContainer));
+            }
+        }
+        else if (item.expectFileInResults())
+        {
+            expectedResults.add(Locator.linkContainingText("/@files"));
+        }
+    }
+
+    public void verifyNoSearchResults()
+    {
+        SearchAdminAPIHelper.waitForIndexer();
+
+        Map<String, SearchResultsQueue.SearchItem> queuedItems = _searchResultsQueue.getQueuedItems();
+        SearchResultsQueue noResultsQueue = new SearchResultsQueue();
+        TestLogger.log("Verify empty search results for previously queued items.");
+        for (String searchTerm : queuedItems.keySet())
+        {
+            noResultsQueue.enqueueSearchItem(searchTerm);
+        }
+        verifySearchResults(null, "noResults", noResultsQueue, maxTries);
+    }
+
+    public void assertNoSearchResult(String searchTerm)
+    {
+        long startTime = System.currentTimeMillis();
+        List<WebElement> results;
+
+        do {
+            SearchResults searchResultsPage = searchFor(searchTerm);
+            results = searchResultsPage.getResults();
+        } while (System.currentTimeMillis() - startTime < BaseWebDriverTest.WAIT_FOR_PAGE && !results.isEmpty());
```

Comment:
I realize sleep is not ideal, but might this be a case for it? Just sleep for a period of time, then do the check, rather than have a tight loop like this.